relays/ethereum/src/rpc.rs

```diff
@@ -0,0 +1,338 @@
+		let params = Params::Array(vec![account]);
+
+		Ok(Substrate::system_account_next_index(&mut self.client, params).await?)
+	}
+
+	/// Returns best Ethereum block that Substrate runtime knows of.
+	async fn best_ethereum_block(&mut self) -> Result<EthereumHeaderId> {
+		let call = EthereumHeadersApiCalls::BestBlock.to_string();
+		let data = "0x".to_string();
+		let params = Params::Array(vec![serde_json::Value::String(call), serde_json::Value::String(data)]);
+
+		let encoded_response = Substrate::state_call(&mut self.client, params).await?;
+		let decoded_response: (u64, sp_bridge_eth_poa::H256) = Decode::decode(&mut &encoded_response.0[..])?;
+
+		let best_header_id = HeaderId(decoded_response.0, decoded_response.1);
+		Ok(best_header_id)
+	}
+
+	/// Returns whether or not transactions receipts are required for Ethereum header submission.
+	// Should I work with a QueuedEthereumHeader or a SubstrateEthereumHeader since that's what'll
+	// actually get encoded and sent to the RPC?
+	//
+	// I'm leaning towards the SubstrateEthereumHeader since that's a bit "lower level"
+	async fn ethereum_receipts_required(&mut self, header: SubstrateEthereumHeader) -> Result<bool> {
+		let call = EthereumHeadersApiCalls::IsImportRequiresReceipts.to_string();
+		let data = Bytes(header.encode());
+		let params = Params::Array(vec![serde_json::Value::String(call), serde_json::to_value(data)?]);
+
+		let encoded_response = Substrate::state_call(&mut self.client, params).await?;
+		let receipts_required: bool = Decode::decode(&mut &encoded_response.0[..])?;
+
+		// Gonna make it the responsibility of the caller to return (receipts_required, id)
+		Ok(receipts_required)
+	}
+
+	/// Returns whether or not the given Ethereum header is known to the Substrate runtime.
+	// The Substrate module could prune old headers. So this function could return false even
+	// if header is synced. And we'll mark corresponding Ethereum header as Orphan.
+	//
+	// But when we read the best header from Substrate next time, we will know that
+	// there's a better header. This Orphan will either be marked as synced, or
+	// eventually pruned.
+	async fn ethereum_header_known(&mut self, header_id: EthereumHeaderId) -> Result<bool> {
+		let call = EthereumHeadersApiCalls::IsKnownBlock.to_string();
+		let data = Bytes(header_id.1.encode());
+		let params = Params::Array(vec![serde_json::Value::String(call), serde_json::to_value(data)?]);
+
+		let encoded_response = Substrate::state_call(&mut self.client, params).await?;
+		let is_known_block: bool = Decode::decode(&mut &encoded_response.0[..])?;
+
+		// Gonna make it the responsibility of the caller to return (is_known_block, id)
+		Ok(is_known_block)
+	}
+
+	/// Submit an extrinsic for inclusion in a block.
+	// Q: Should I move the UncheckedExtrinsic type elsewhere so I don't have to pull it in from
+	// the runtime?
+	async fn submit_extrinsic(&mut self, transaction: UncheckedExtrinsic) -> Result<SubstrateHash> {
```

Comment:
IMHO you can change the input here to `Bytes`. I think it's more important to keep the RPC access layer runtime agnostic than to make it super strictly typed.